gossip/src/crds_gossip_pull.rs

```diff
@@ -143,19 +141,26 @@ impl CrdsFilter {
 
 /// A vector of crds filters that together hold a complete set of Hashes.
 struct CrdsFilterSet {
-    filters: Vec<AtomicBloom<Hash>>,
+    filters: Vec<Option<AtomicBloom<Hash>>>,
     mask_bits: u32,
 }
 
 impl CrdsFilterSet {
-    fn new(num_items: usize, max_bytes: usize) -> Self {
+    fn new<R: Rng>(rng: &mut R, num_items: usize, max_bytes: usize) -> Self {
+        const SAMPLE_RATE: usize = 8;
+        const MAX_NUM_FILTERS: usize = 1024;
         let max_bits = (max_bytes * 8) as f64;
         let max_items = CrdsFilter::max_items(max_bits, FALSE_RATE, KEYS);
         let mask_bits = CrdsFilter::mask_bits(num_items as f64, max_items);
-        let filters =
-            repeat_with(|| Bloom::random(max_items as usize, FALSE_RATE, max_bits as usize).into())
-                .take(1 << mask_bits)
-                .collect();
+        let mut filters: Vec<_> = repeat_with(|| None).take(1usize << mask_bits).collect();
+        let mut indices: Vec<_> = (0..filters.len()).collect();
+        let size = (filters.len() + SAMPLE_RATE - 1) / SAMPLE_RATE;
+        for _ in 0..MAX_NUM_FILTERS.min(size) {
+            let k = rng.gen_range(0..indices.len());
+            let k = indices.swap_remove(k);
+            let filter = Bloom::random(max_items as usize, FALSE_RATE, max_bits as usize);
+            filters[k] = Some(AtomicBloom::<Hash>::from(filter));
+        }
```

Comment:
are you just sending fewer bloom filters to get populated by your neighbor? now `filters` vec is populated with only some Bloom filters? essentially replacing this logic in `new_pull_request()`: 
```
        let mut filters = self.build_crds_filters(thread_pool, crds, bloom_size);
        if filters.len() > MAX_NUM_PULL_REQUESTS {
            for i in 0..MAX_NUM_PULL_REQUESTS {
                let j = rng.gen_range(i..filters.len());
                filters.swap(i, j);
            }
            filters.truncate(MAX_NUM_PULL_REQUESTS);
        }
```

Reply:
yeah, that is basically what the commit is doing.